insights/parsers/du_usage.py

```diff
@@ -0,0 +1,91 @@
+"""
+du - Command
+============
+
+Module for the processing of output from the ``du`` command.  The base class
+``DiskUsage`` provides all of the functionality for all classes.
+
+Sample input data for the ``du -sh`` command looks like::
+22G     /var/lib/mongodb
+
+This example demonstrates the ``DiskUsage`` class but all classes will provide
+the same functionality.
+
+Examples:
+    >>> du_info = shared[DiskUsage_SH]
+    >>> du_info.get_size
+    {'/var/lib/mongodb': '22G'}
+
+"""
+
+from .. import Parser, parser, LegacyItemAccess
+from .. import get_active_lines
+
+
+def parse_du_lines(du_content):
+    """Parse contents of each line in ``du`` output.
+
+    Parameters:
+        du_content : Lines of df output to be parsed.
+
+    Returns:
+        Dict : A dict of key pair values where key is filesystem path
+        and value is the size of filesystem.
+    """
+    data = {}
+    for line in get_active_lines(du_content, comment_char="COMMAND>"):
+        size, path = line.split()
+        data[path] = size
+    return data
+
+
+@parser("du_-h")
+class DiskUsage(LegacyItemAccess, Parser):
+    """Class to provide methods used by all ``df`` command classes.
+
+    Attributes:
+        data: List of output of du command which contains filesystem and
+        estimate file space usage.
+
+    """
+    def __len__(self):
+        return len(self.data)
+
+    def parse_content(self, content):
+        self.data = parse_du_lines(content)
+        return self.data
+
+    def get_size(self, dir_path):
+        return self.data.get(dir_path)
+
+
+@parser("du_-sh")
+class DiskUsage_SH(DiskUsage):
```

Comment:
There is no spec "du_-sh" anymore.
I think this class can be removed.

Reply:
removed.

Reply:
Removed now.